doc/decisions/snippet_sharing_implementation_details.md

```diff
@@ -0,0 +1,27 @@
+# Implementation details & Technical description
+
+## REST service
+
+The REST service will be implemented using [CppCMS](http://cppcms.com/wikipp/en/page/main). A detailed description of the implemented API can be found in the [API description](api-description.apib). The REST service will leave some space for configuration, as not every deployment will look the same.
+
+## Front-end
+
+The front-end will be implemented using AngularJS (v1.5), i.e. a SPA (Single Page Application) will be developed. The final SPA will only consist of static files (.html, .js, .css, .jpg, ...), which means that it could be deployed within any web server. To build the SPA from the development resources, [Grunt](http://gruntjs.com/) is used as task runner.
+
+### Elektra Website
+
+One part of the front-end will be the new Elektra website containing documentation, tutorials and other important artifacts. If it is possible to display the documentation artifacts without changing their current arrangement (i.e. rename them on GitHub), it would also be easily possible to implement per-version documentation.
+
+Resources will be fetched directly from GitHub. It is planned to load all content of the website from GitHub, to avoid having to change the front-end itself. The front-end will support Markdown syntax via a plugin like [Showdown](https://github.com/showdownjs/showdown). Styling can then be done easily through CSS.
```

Comment:
There is a small service class taking care of retrieving the docs and for the URLs we have some constants. I think a change to bitbucket for example wouldn't be a huge hassle (no idea at all if they also offer an API).